### ISSUE TITLE
Backend on IBMQ devices 

### ISSUE THREAD

**Issue by ycchen1989** [XanaduAI/pennylane#227]
Although it is possible to run PennyLane program on IBMQ public version, it seems currently not able to run on IBMQ services that are exclusively to institution subscriptions!
According to the attached link, when we want to access the institute-exclusive IBMQ services, it is not only token but also URL are needed to login. Is it possible to extend this capability to the plugin?
ref:
https://qiskit.org/documentation/install.html#access-ibm-q-systems

**Comment by quantshah** [comment]
Hi, are you using this plugin: [https://github.com/carstenblank/pennylane-qiskit](https://github.com/carstenblank/pennylane-qiskit)? It needs some updates and I am looking into it.

They have instructions on how to use your token to run the institute exclusive services:

> Tests on the IBMQ device can only be run if a ibmqx_token for the IBM Q experience is configured in the PennyLane configuration file. If this is the case, running make test also executes tests on the ibmq device. By default tests on the ibmq device run with ibmq_qasm_simulator backend and those done by the basicaer and aer device are run with the qasm_simulator backend. At the time of writing this means that the test are "free". Please verify that this is also the case for your account.

You can see the documentation here:
[https://pennylane-qiskit.readthedocs.io/en/latest/gettingstarted.html](https://pennylane-qiskit.readthedocs.io/en/latest/gettingstarted.html)

Here is a code snippet

```
import pennylane as qml
dev = qml.device('qiskit.ibmq', wires=2, ibmqx_token="XXX")
```

Please let me know if this works as I am trying to update the plugin.

**Comment by quantshah** [comment]
We could have something like:

```
dev = qml.device('qiskit.ibmq', wires=2, ibmqx_token="XXX", url="xyz.ibm")
```

and make a check in the __init__ for the device being "qiskit.ibmq".

**Comment by ycchen1989** [comment]
From IBMQ document, the institute exclusive service should login in the following format:
`
from qiskit import IBMQ
IBMQ.save_account('MY_API_TOKEN', 'MY_URL')
`
it seems there is currently no fields in pennylane-qiskit to fill the "MY_URL".

**Comment by co9olguy** [comment]
Thanks @ycchen1989 for the report of this missing functionality. We will work with IBM to get the appropriate information necessary to remedy it as soon as possible.

**Comment by josh146** [comment]
Hi @ycchen1989. Apologies, I'm not so familiar with the IBMQ institutional access. I had a look on the link you posted, and it looks like once you save the token and URL in your Qiskit configuration, it should be automatically used the next time Qiskit has loaded:

![Screenshot_20190611_113038](https://user-images.githubusercontent.com/2959003/59285675-58f6b380-8c3c-11e9-8b4e-9e659dc7ffe0.png)

Just confirming that you have followed the steps above, and then used PennyLane-qiskit to run an IBMQ device? Does this still not work (and if not, what error message do you get?)

**Comment by quantshah** [comment]
Also, I think you have to enable the account:

```
IBMQ.enable_account("token")
```

Right now I am facing some issue with connecting to the server even though I have accepted the license online.

```
ConnectionError: Couldn't connect to IBMQ server: error during login: License required. You need to accept the License from Web Platform. https://quantumexperience.ng.bluemix.net
```

**Comment by quantshah** [comment]
It seems after following these steps, one can use the new URL login API.

1. Install qiskit and run the followig code (before runnign the pennylane plugin) to enable your token and url on the machine (I still do not know where IBM stores these).

```
from qiskit import IBMQ
IBMQ.save_account(token, url)
IBMQ.enable_account(token)
```

2. Go to the folder from which you want to run pennylane-qiskit code and set the path variable IBMQX_TOKEN as, 

```
export IBMQX_TOKEN=token
```
This should make the pennylane-qiskit plugin work. I am able to successfully connect and run the basic tests.

If you get some error like me regarding the license. Then goto: [https://quantumexperience.ng.bluemix.net](https://quantumexperience.ng.bluemix.net) and accept their license agreement. 

I hope this solves this issue, @ycchen1989 please let us know if this helps and I can add this bit to the documentation of the pennylane qiskit plugin.

**Comment by ycchen1989** [comment]
> It seems after following these steps, one can use the new URL login API.
> 
>     1. Install qiskit and run the followig code (before running the pennylane plugin) to enable your token and url on the local machine (I still do not know where qiskit stores these).
> 
> 
> ```
> from qiskit import IBMQ
> IBMQ.save_account(token, url)
> IBMQ.enable_account(token)
> ```
> 
>     1. Go to the folder from which you want to run pennylane-qiskit code and set the path variable IBMQX_TOKEN as,
> 
> 
> ```
> export IBMQX_TOKEN=token
> ```
> 
> This should make the pennylane-qiskit plugin work. I am able to successfully connect and run the basic tests.
> 
> If you get some error like me regarding the license. Then goto https://quantumexperience.ng.bluemix.net and accept their license agreement.
> 
> I hope this solves this issue, @ycchen1989 please let us know if this helps and I can add this bit to the documentation of the pennylane qiskit plugin.

I just tried this. However, when I ran the first part to save the account, it show the license issue which should not be because I already clicked such things in the IBMQ web console. I am wondering whether it send authentication to correct remote servers since that message do not appear when I use pure qiskit.

The following is the code I used with my pure-qiskit code and can run without any issue.
```
IBMQ.enable_account(token, url)
```

**Comment by quantshah** [comment]
> I just tried this. However, when I ran the first part to save the account, it show the license issue which should not be because I already clicked such things in the IBMQ web console. I am wondering whether it send authentication to correct remote servers since that message do not appear when I use pure qiskit.

Even I faced this issue. There was a different (perhaps older site) where you had to login and accept it:
https://quantumexperience.ng.bluemix.net

Could you try this and see if it works. It did for my case.

**Comment by johannesjmeyer** [comment]
Interestingly, I encountered the same problem when using the Qiskit plugin yesterday, and as @quantshah mentioned, this seems to be due to the new IBM Q Experience platform. But if I remember it correctly, then calling `IBMQ.save_account('MY_API_TOKEN', 'MY_URL')` before I load the device did the job.

I think this could be solved by simply introducing a kwarg for the url and an entry for it in the configuration file.

**Comment by josh146** [comment]
> I think this could be solved by simply introducing a kwarg for the url and an entry for it in the configuration file.

Is this now fixed in XanaduAI/pennylane-qiskit#39?

**Comment by josh146** [comment]
This should now be fixed in XanaduAI/pennylane-qiskit#44
